packages/block-library/src/query/edit/inspector-controls/index.js

```diff
@@ -123,6 +124,18 @@ export default function QueryInspectorControls( props ) {
 		isControlAllowed( allowedControls, 'parents' ) &&
 		isPostTypeHierarchical;
 
+	const enhancedPaginationNotice = __(
+		'Enhanced Pagination might cause interactive blocks within the Post Template to stop working. Disable it if you experience any issues.'
+	);
+
+	const isFirstRender = useRef( true ); // Don't speak on first render.
+	useEffect( () => {
+		if ( ! isFirstRender.current && enhancedPagination ) {
+			speak( enhancedPaginationNotice );
+		}
+		isFirstRender.current = false;
+	}, [ enhancedPagination, enhancedPaginationNotice ] );
```

Comment:
I need help understanding this part. 😅 This is supposed to prevent announcing the risks associated with the Enhanced Pagination if it's already enabled during the first render, am I right?

Shouldn't we follow the same logic for the `Notice` component rendered below?

If that's the case, maybe we could just simplify the code so we turn `isFirstRender` into regular state and use it along with `enhancedPagination` to conditionally render the `Notice` component, passing the `enhancedPaginationNotice` value as the `spokenMessage`. 🤔

Reply:
> Shouldn't we follow the same logic for the `Notice` component rendered below?

No, the notice should appear on the screen, but it should not be announced if the setting is already enabled.

It's basically the same approach that we used here:

- https://github.com/WordPress/gutenberg/pull/52251